### PR TITLE
Mes-9336 | GA4 test report cat b

### PR DESCRIPTION
**Pull request by gittins7**
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/pages/test-report/test-report.analytics.effects.ts [hunk 1]
```diff
@@ -633,12 +769,20 @@ export class TestReportAnalyticsEffects {
...
       [ReturnType<typeof controlledStopActions.ControlledStopAddDangerousFault>, TestsModel, boolean],
     ) => {
+      // TODO MES-9495 - remove old analytics
       this.analytics.logEvent(
         formatAnalyticsText(AnalyticsEventCategories.TEST_REPORT, tests),
         formatAnalyticsText(AnalyticsEvents.ADD_DANGEROUS_FAULT, tests),
         fullCompetencyLabels['outcomeControlledStop'],
         1,
       );
+      this.analytics.logGAEvent(
```
Comment (gittins7) [inline src/app/pages/test-report/test-report.analytics.effects.ts:781]: Missing GA4 Comment

File: src/app/pages/test-report/test-report.analytics.effects.ts [hunk 2]
```diff
@@ -1032,11 +1281,19 @@ export class TestReportAnalyticsEffects {
       [action, tests]:
       [ReturnType<typeof vehicleChecksActions.ShowMeQuestionRemoveFault>, TestsModel, boolean],
     ) => {
+      // TODO MES-9495 - remove old analytics
       this.analytics.logEvent(
         formatAnalyticsText(AnalyticsEventCategories.TEST_REPORT, tests),
         formatAnalyticsText(CompetencyOutcomeAnalyticEvent(action.faultLevel), tests),
         fullCompetencyLabels.showMeQuestion,
       );
+      this.analytics.logGAEvent(
```
Comment (gittins7) [inline src/app/pages/test-report/test-report.analytics.effects.ts:1293]: Missing GA4 Comment

File: src/app/pages/test-report/test-report.analytics.effects.ts [hunk 3]
```diff
@@ -1440,11 +1776,19 @@ export class TestReportAnalyticsEffects {
       [, tests]:
       [ReturnType<typeof manoeuvresADIPart2Actions.RecordManoeuvresSelection>, TestsModel, boolean],
     ) => {
+      // TODO MES-9495 - remove old analytics
       this.analytics.logEvent(
         formatAnalyticsText(AnalyticsEventCategories.TEST_REPORT, tests),
         formatAnalyticsText(AnalyticsEvents.TOGGLE_LEGAL_REQUIREMENT, tests),
         `${legalRequirementsLabels['manoeuvre']} - ${legalRequirementToggleValues.completed}`,
       );
+      this.analytics.logGAEvent(
```
Comment (gittins7) [inline src/app/pages/test-report/test-report.analytics.effects.ts:1789]: Missing GA4 Comment

File: src/app/pages/test-report/test-report.analytics.effects.ts [hunk 4]
```diff
@@ -1562,12 +1933,20 @@ export class TestReportAnalyticsEffects {
...
       [ReturnType<typeof uncoupleRecoupleActions.UncoupleRecoupleAddDrivingFault>, TestsModel, boolean],
     ) => {
+      // TODO MES-9495 - remove old analytics
       this.analytics.logEvent(
         formatAnalyticsText(AnalyticsEventCategories.TEST_REPORT, tests),
         formatAnalyticsText(AnalyticsEvents.ADD_DRIVING_FAULT, tests),
         'Uncouple recouple',
         1,
       );
+      this.analytics.logGAEvent(
```
Comment (gittins7) [inline src/app/pages/test-report/test-report.analytics.effects.ts:1948]: Missing GA4 Comment

File: src/app/pages/test-report/test-report.analytics.effects.ts [hunk 5]
```diff
@@ -2334,10 +2731,15 @@ export class TestReportAnalyticsEffects {
       [, tests]:
       [ReturnType<typeof testReportActions.StartTimer>, TestsModel, boolean],
     ) => {
+
```
Comment (gittins7) [inline src/app/pages/test-report/test-report.analytics.effects.ts]: Missing TODO

File: src/app/shared/helpers/competency-outcome-analytic-event.ts [hunk 6]
```diff
@@ -11,3 +11,14 @@ export const CompetencyOutcomeAnalyticEvent = (faultLevel: CompetencyOutcome): s
       return AnalyticsEvents.REMOVE_DRIVING_FAULT;
   }
 };
+
+export const CompetencyOutcomeGA4Event = (faultLevel: CompetencyOutcome): string => {
```
Comment (andrewsetterfield) [inline src/app/shared/helpers/competency-outcome-analytic-event.ts:16]: unit test please.  plus can you reduce the duplication and use this enum from fault-comments

enum ValidFaultTypes {
  DRIVING = 'driving',
  SERIOUS = 'serious',
  DANGEROUS = 'dangerous',
}

File: src/app/pages/test-report/test-report.analytics.effects.ts [hunk 7]
```diff
@@ -259,12 +293,21 @@ export class TestReportAnalyticsEffects {
...
         fullCompetencyLabels[action.faultPayload.competency],
         action.faultPayload.newFaultCount,
       );
+      // GA4 Analytics
+      this.analytics.logGAEvent(
+        formatAnalyticsText(GoogleAnalyticsEvents.ADD_FAULT, tests),
+        formatAnalyticsText(GoogleAnalyticsEventsTitles.FAULT_TYPE, tests),
+        fullCompetencyLabels[action.faultPayload.competency],
+        formatAnalyticsText(GoogleAnalyticsEventsTitles.SEVERITY, tests),
+        formatAnalyticsText(GoogleAnalyticsEventsValues.DRIVING, tests),
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/test-report.analytics.effects.ts]: please use the common enum for references to driving/serious/dangerous.  that goes for all references in this file

enum ValidFaultTypes

**Review by andrewsetterfield — APPROVED** [review]

**Review by RLCorp — APPROVED** [review]

**Pull request merged**